indexbuilder.go

```diff
@@ -425,6 +429,16 @@ func (b *IndexBuilder) Add(doc Document) error {
 		}
 	}
 
+	// only compute language metadata for ambiguous extensions
+	if doc.Language == "" && len(enry_data.LanguagesByExtension[path.Ext(doc.Name)]) >= 2 {
```

Comment:
can't we just always use enry.GetLanguage?

Reply:
No. Not easily, at least. The LanguageMap is limited to 256 entries, which is fine for ctags limited support, but enry recognizes more than that, so a large repo with sample code will fail to index because of too many different filetypes.

Most of the complexity stems from this limitation.